osquery/tables/system/linux/block_devices.cpp

```diff
@@ -9,20 +9,56 @@
  */
 
 #include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
 
 #include <blkid/blkid.h>
 #include <libudev.h>
+#include <linux/kdev_t.h>
 #include <unistd.h>
 
 #include <osquery/core.h>
 #include <osquery/filesystem.h>
 #include <osquery/logger.h>
 #include <osquery/tables.h>
 
+extern "C" {
+#include <lvm2app.h>
+}
+
 namespace osquery {
 namespace tables {
 
-static void getBlockDevice(struct udev_device *dev, QueryData &results) {
+void populatePVChildren(lvm_t lvm,
+                        const std::string& devname,
+                        const std::string& pvid,
+                        std::map<std::string, std::string>& lvm_lv2pv) {
+  const auto pvid_copy = boost::erase_all_copy(pvid, "-");
+  const char* vg_name = lvm_vgname_from_pvid(lvm, pvid_copy.c_str());
+  if (vg_name == nullptr) {
+    return;
+  }
+
+  vg_t vg = lvm_vg_open(lvm, vg_name, "r", 0);
```

Comment:
It looks like this could potentially return a `nullptr`: http://lvm2.sourcearchive.com/documentation/2.02.88-1/classlvm__t_a4784fe8a27d64b5c8e802a57c3e1ec35.html